ui/packages/consul-peerings/app/components/consul/peer/form/generate/index.hbs

```diff
@@ -0,0 +1,56 @@
+<div
+  class={{class-map
+    'consul-peer-form-generate'
+  }}
+  ...attributes
+>
+  <StateMachine
+    @src={{require './chart.xstate' from="/components/consul/peer/form/generate"}}
+    @initial={{if @regenerate 'loading' 'idle'}}
+  as |fsm|>
+
+  {{#let
+    (unique-id)
+  as |id reset|}}
+    <form
+      {{on 'submit' (fn fsm.dispatch 'LOAD')}}
+      id={{id}}
+    >
+
+      <fsm.State @matches={{array 'idle'}}>
+        {{yield (hash
+          Fieldsets=(component "consul/peer/form/generate/fieldsets"
+            item=@item
+          )
+          Actions=(component "consul/peer/form/generate/actions"
+            item=@item
+            id=id
+          )
+        )}}
+      </fsm.State>
+
+      <fsm.State @matches={{'loading'}}>
+        <DataSource
+          @src={{uri '/${partition}/${nspace}/${dc}/peering/token-for/${name}'
+            (hash
+              partition=@item.Partition
+              nspace=''
+              dc=@item.Datacenter
+              name=@item.Name
+            )
+          }}
+          @onchange={{queue
+            @onchange
+            (pick 'data' (fn fsm.dispatch 'SUCCESS'))
+          }}
+        />
+      </fsm.State>
+
+      <fsm.State @matches={{'success'}}>
```

Comment:
do we want to handle success here? 

There doesn't seem to be a section for the error state in this template. do we want to add handling for the error state here as well?

Reply:
Ah this was to PR this in small pieces sorry!

https://github.com/hashicorp/consul/pull/13755 add this.